File: app/src/main/java/com/example/myappcontacts/business/contact/IContactsInteractor.java
package com.example.myappcontacts.business.contact;

import com.example.myappcontacts.data.dao.contacts.db.ContactsModel;

import java.util.UUID;

import io.reactivex.Completable;
import io.reactivex.Single;

public interface IContactsInteractor {
    Single<ContactsModel> loadContact(UUID contactId);
    Completable updateContact(ContactsModel contactsModel);
}

File: app/src/main/java/com/example/myappcontacts/business/contact/IContactsInteractor.kt
package com.example.myappcontacts.business.contact

import com.example.myappcontacts.data.dao.contacts.db.ContactsModel
import io.reactivex.Completable
import io.reactivex.Single
import java.util.*

interface IContactsInteractor {

    fun loadContact(contactId: UUID): Single<ContactsModel>

    fun updateContact(contactsModel: ContactsModel): Completable

}